elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStore.java

```diff
@@ -74,6 +76,11 @@ public void populateEntityDictionary(EntityDictionary dictionary) {
 
     @Override
     public DataStoreTransaction beginTransaction() {
-        return new AggregationDataStoreTransaction(queryEngine, cache);
+        return new AggregationDataStoreTransaction(queryEngine, cache, queryLogger);
+    }
+
+    public AggregationDataStore setupCustomQueryLogger(QueryLogger logger) {
```

Comment:
Why not just set the logger in the constructor?

Reply:
For navi or any other external application to instantiate their own querylogger, this function can be used to update the default QueryLogger from Noop to their custom querylogger.

Reply:
That should happen in the constructor though - not a set function.  Ideally, most stuff should be immutable.